warehouse/schema.go

```diff
@@ -1,53 +1,90 @@
 package warehouse
 
-import "fmt"
+import (
+	"reflect"
+	"time"
+)
+
+type Field interface {
+	Name() string
+	DataType() reflect.Type
+	IsTime() bool
+}
+
+type dataField struct {
+	name     string
+	dataType reflect.Type
+	isTime   bool
+}
 
-type Field struct {
-	Name   string
-	DBType string
+func (f *dataField) Name() string {
+	return f.name
 }
 
-func (f Field) String() string {
-	return fmt.Sprintf("%s %s", f.Name, f.DBType)
+func (f *dataField) DataType() reflect.Type {
+	return f.dataType
+}
+
+func (f *dataField) IsTime() bool {
+	return f.dataType.Kind() == reflect.Struct && f.dataType.PkgPath() == "time" && f.dataType.Name() == "Time"
```

Comment:
Why wouldn't this be `return (reflect.TypeOf(f.dataType) == reflect.TypeOf(time.Time))`?

Reply:
This is just my relative ignorance of reflection in Go, I've updated the check in the newly added `newDataField()` function.